File: javaexamples/src/main/java/com/inappstory/javaexamples/userchange/UserChangeSample.java
package com.inappstory.javaexamples.userchange;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatEditText;

import com.inappstory.javaexamples.DemoApplication;
import com.inappstory.javaexamples.R;
import com.inappstory.javaexamples.favorites.StoryFavoritesActivity;
import com.inappstory.sdk.AppearanceManager;
import com.inappstory.sdk.InAppStoryManager;
import com.inappstory.sdk.stories.ui.list.StoriesList;

import java.util.UUID;


public class UserChangeSample extends AppCompatActivity {

    AppearanceManager appearanceManager = new AppearanceManager();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user);
        String newUser = UUID.randomUUID().toString().substring(0, 8);
        ((AppCompatEditText)findViewById(R.id.userId)).setText(newUser);
        findViewById(R.id.change_user).setOnClickListener(v -> {
            InAppStoryManager.getInstance().setUserId(((AppCompatEditText)findViewById(R.id.userId)).getText().toString());
        });
        showStories();
    }

    @Override
    protected void onDestroy() {
        InAppStoryManager.getInstance().setUserId(DemoApplication.getUserId());
        super.onDestroy();
    }

    private void showStories() {

        StoriesList storiesList = findViewById(R.id.stories_list);
        storiesList.setAppearanceManager(appearanceManager.csHasFavorite(true));
        storiesList.setOnFavoriteItemClick(() -> {
            Intent intent = new Intent(UserChangeSample.this, StoryFavoritesActivity.class);
            startActivity(intent);
        });
        storiesList.loadStories();
    }

}

File: kotlinexamples/src/main/java/com/inappstory/kotlinexamples/userchange/UserChangeSample.kt
package com.inappstory.kotlinexamples.userchange

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatEditText
import com.inappstory.kotlinexamples.DemoApplication
import com.inappstory.kotlinexamples.R
import com.inappstory.kotlinexamples.favorites.StoryFavoritesActivity
import com.inappstory.sdk.AppearanceManager
import com.inappstory.sdk.InAppStoryManager
import com.inappstory.sdk.stories.outercallbacks.common.errors.ErrorCallbackAdapter
import com.inappstory.sdk.stories.ui.list.StoriesList
import java.util.*

class UserChangeSample : AppCompatActivity() {
    private var appearanceManager: AppearanceManager = AppearanceManager()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_user)
        InAppStoryManager.getInstance().setErrorCallback(object : ErrorCallbackAdapter() {
            override fun sessionError() {
                runOnUiThread {
                    Toast.makeText(
                        this@UserChangeSample,
                        "Can't open session",
                        Toast.LENGTH_LONG
                    ).show()
                }
            }
        })

        findViewById<View>(R.id.change_user).setOnClickListener {
            var userId: String? =
                (findViewById<View>(R.id.userId) as AppCompatEditText).text.toString()
            if (userId.isNullOrEmpty()) userId = null
            var userSign: String? =
                (findViewById<View>(R.id.userSign) as AppCompatEditText).text.toString()
            if (userSign.isNullOrEmpty()) userSign = null
            userId?.let {
                InAppStoryManager.getInstance().setUserId(it, userSign)
            }
        }
        showStories()
    }

    override fun onDestroy() {
        InAppStoryManager.getInstance().setUserId(DemoApplication.getUserId(), null)
        super.onDestroy()
    }

    private fun showStories() {
        val storiesList = findViewById<StoriesList>(R.id.stories_list)
        storiesList.setAppearanceManager(appearanceManager.csHasFavorite(true))
        storiesList.setOnFavoriteItemClick {
            val intent = Intent(this@UserChangeSample, StoryFavoritesActivity::class.java)
            startActivity(intent)
        }
        storiesList.loadStories()
    }
}